share/genbuild.sh

```diff
@@ -18,17 +18,14 @@ else
     exit 1
 fi
 
-GIT_TAG=""
-GIT_COMMIT=""
+GIT_TAG="${GIT_TAG:-}"
+GIT_COMMIT="${GIT_COMMIT:-}"
 if [ "${BITCOIN_GENBUILD_NO_GIT}" != "1" ] && [ -e "$(command -v git)" ] && [ "$(git rev-parse --is-inside-work-tree 2>/dev/null)" = "true" ]; then
     # clean 'dirty' status of touched files that haven't been modified
     git diff >/dev/null 2>/dev/null
 
     # if latest commit is tagged and not dirty, then override using the tag name
-    RAWDESC=$(git describe --abbrev=0 2>/dev/null)
-    if [ "$(git rev-parse HEAD)" = "$(git rev-list -1 $RAWDESC 2>/dev/null)" ]; then
-        git diff-index --quiet HEAD -- && GIT_TAG=$RAWDESC
-    fi
+    git diff-index --quiet HEAD -- && GIT_TAG=$(git describe --exact-match 2>/dev/null)
```

Comment:
6b6b8e7edeba3360738fb40ebc12872a81089967

Why this change is required?

Reply:
It isn't, but it's simpler.